src/api/che-login-manager.ts

```diff
@@ -0,0 +1,529 @@
+/*********************************************************************
+ * Copyright (c) 2020 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import axios, { AxiosInstance } from 'axios'
+import * as fs from 'fs'
+import * as https from 'https'
+import * as jwt from 'jsonwebtoken'
+import * as path from 'path'
+import * as querystring from 'querystring'
+
+import { ACCESS_TOKEN_KEY } from '../common-flags'
+
+import { CheApiClient } from './che-api-client'
+
+// Represents login information to use for requests
+// Notice: accessToken is undefined for single user mode
+export interface LoginData {
+  cheApiEndpoint: string
+  accessToken: string | undefined
+}
+
+// Credentials file format
+export interface CheServerLoginConfig {
```

Comment:
I would add some field like `version` in case we change the format

Reply:
Well, I hope it will not be needed, but yes, we don't know for sure. Will add the field.